src/App.tsx

```diff
@@ -1,24 +1,185 @@
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
-  );
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [removeTodoIsClicked, setRemoveTodoIsClicked] = useState(false);
+  const [editTodoIsClicked, setEditTodoIsClicked] = useState(false);
+  const [filterMode, setFilterMode] = useState<FilterType>(FilterType.All);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [fetchTodosStatus, setFetchTodosStatus] = useState(false);
+  const [onEmptyFormSubmit, setOnEmptyFormSubmit] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [toggleTodoIsFailed, setToggleTodoIsFailed] = useState(false);
+  const [updatingTogosToggle, setUpdatingTodosToggle] = useState<number[]>([]);
+  const [deletingCompletedTodos,
+    setDeletingCompletedTodos] = useState<number[]>([]);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+  const allTodosCompleted = useMemo(() => {
+    return filteredTodos.every((todo) => todo.completed);
+  }, [filteredTodos]);
+
+  const notAllTodosCompleted = useMemo(() => {
+    return filteredTodos.every((todo) => todo.completed);
+  }, [filteredTodos]);
+
+  const showNotification = removeTodoIsClicked
+  || editTodoIsClicked
+  || onEmptyFormSubmit
+  || toggleTodoIsFailed;
+
+  const getTodos = async () => {
+    try {
+      const data = await client.get(`/todos?userId=${USER_ID}`) as Todo[];
+
+      setTodos(data);
+      setFetchTodosStatus(true);
+    } catch {
+      setFetchTodosStatus(false);
+    }
+  };
+
+  useEffect(() => {
+    getTodos();
+  }, []);
+
+  const filteredTodosArr = useMemo(() => {
+    return todos.filter((todo) => {
+      if (filterMode === FilterType.Active) {
+        return !todo.completed;
+      }
+
+      if (filterMode === FilterType.Completed) {
+        return todo.completed;
+      }
+
+      return todo;
+    });
+  }, [todos, filterMode]);
+
+  useEffect(() => {
+    setFilteredTodos(filteredTodosArr);
+  }, [filteredTodosArr]);
+
+  const handleToggleAllClick = async () => {
+    const updatedTodos = filteredTodosArr.map((todo) => ({
+      ...todo,
+      completed: !allTodosCompleted,
+    }));
+
+    try {
+      setUpdatingTodosToggle(updatedTodos.map((todo) => todo.id));
+
+      await Promise.all(
+        updatedTodos.map(async (todo) => {
+          await fetch(`https://mate.academy/students-api/todos/${todo.id}`,
+            {
+              method: 'PATCH',
+              headers: {
+                'Content-Type': 'application/json;charset=utf-8',
+              },
+              body: JSON.stringify({
+                completed: todo.completed,
+              }),
+            });
+        }),
+      );
+      setTodos(updatedTodos);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log(`An error occurred while toggling all todos ${error}`);
```

Comment:
Don't use console.log in production code. You should create an error handling system to show user-friendly messages when errors occur.

Reply:
I have just removed the console part as 
`setToggleTodoIsFailed(true); `
already triggers displaying of the 'unable to update a todo' error